src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -200,7 +199,42 @@ public async Task<string> SubmitPledge(CreatePostRequest request)
 
             return _encodingService.Encode(pledgeId, EncodingType.PledgeId);
         }
+        public async Task RejectApplications(RejectApplicationsPostRequest request)
+        { 
+            var serviceRequest = new SetRejectApplicationsRequest
+            {
+                UserId = _userService.GetUserId(),
+                UserDisplayName = _userService.GetUserDisplayName(),
+            };
+            serviceRequest.ApplicationsToReject.AddRange(from applicationId in request.ApplicationsToReject
+                                                         select (int)_encodingService.Decode(applicationId,
+                                                         EncodingType.PledgeApplicationId));
+
+            await _pledgeService.RejectApplications(serviceRequest, request.AccountId, request.PledgeId);
+        }
 
+        public async Task<RejectApplicationsViewModel> GetRejectApplicationsViewModel(RejectApplicationsRequest request)
+        {
+            var rejectApplicationsViewModel = new RejectApplicationsViewModel
+            {
+                EncodedAccountId = request.EncodedAccountId,
+                EncodedPledgeId = request.EncodedPledgeId
+            };
+
+            var applicationsList = await _pledgeService.GetApplicationsDasNames(request.AccountId, request.PledgeId);
```

Comment:
As above, could we create the object in a single statement and avoid this LINQ?

I think the whole method could be rewritten as:
```
            var applicationsList = await _pledgeService.GetApplicationsDasNames(request.AccountId, request.PledgeId);

            var rejectApplicationsViewModel = new RejectApplicationsViewModel
            {
                EncodedAccountId = request.EncodedAccountId,
                EncodedPledgeId = request.EncodedPledgeId,
                DasAccountNames = applicationsList.Applications.Where(x =>
                        request.ApplicationsToReject.Any((s => (int)_encodingService.Decode(s, EncodingType.PledgeApplicationId) == x.Id)))
                    .Select(app => app.DasAccountName ).ToList()
            };
            
            return rejectApplicationsViewModel;
```

This is not tested though.

Reply:
I have refactored this method and removed the LINQ as well.